src/presentation/http/router/noteSettings.test.ts

```diff
@@ -281,4 +281,73 @@ describe('NoteSettings API', () => {
 
     test.todo('Return 403 when user authorized, but not member of the team');
   });
+
+  describe('PATCH /note-teams/new-role/:newRole', () => {
+    test('If we want to change role to "write", in database it will be stored as number : 1', async () => {
+      await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/new-role/write',
+        body: {
+          'userId': 1,
+          'noteId': 2,
+        },
+      });
+
+      const team = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/Pq1T9vc23Q/team',
+      });
+
+      if (team?.json() !== undefined) {
+        expect(team?.json()).toStrictEqual([
+          {
+            'id': 2,
+            'noteId': 2,
+            'userId': 1,
+            'role': 1,
+          },
+        ]);
+      }
+    });
+
+    test('Returns status code 200 and new role if it was patched', async () => {
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/new-role/write',
+        body: {
+          'userId': 1,
+          'noteId': 2,
+        },
+      });
+
+      expect(response?.statusCode).toBe(200);
+      expect(response?.body).toBe('write');
+    });
+
+    test('Returns status code 404 and "User in team is not found" message if no such a note exists', async () => {
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/new-role/write',
+        body: {
+          'userId': 15,
+          'noteId': 0,
+        },
+      });
+
+      expect(response?.statusCode).toBe(404);
+      expect(response?.json().message).toBe('User in team not found');
+    });
+  });
```

Comment:
pls cover a case when user already has passing role

Reply:
i guess this case is already covered since we already patched role in previous test